workflows/theiaviral/wf_theiaviral_illumina_pe.wdl

```diff
@@ -0,0 +1,437 @@
+version 1.0
+
+import "../utilities/wf_read_QC_trim_pe.wdl" as read_qc
+import "../../tasks/quality_control/comparisons/task_screen.wdl" as read_screen_task
+import "../../tasks/utilities/task_rasusa.wdl" as rasusa_task
+import "../../tasks/assembly/task_spades.wdl" as spades_task
+import "../../tasks/assembly/task_megahit.wdl" as megahit_task
+import "../../tasks/quality_control/advanced_metrics/task_checkv.wdl" as checkv_task
+import "../../tasks/quality_control/basic_statistics/task_quast.wdl" as quast_task
+import "../../tasks/taxon_id/task_skani.wdl" as skani_task
+import "../../tasks/utilities/data_import/task_ncbi_datasets.wdl" as ncbi_datasets_task
+import "../../tasks/taxon_id/task_identify_taxon_id.wdl" as identify_taxon_id_task
+import "../../tasks/alignment/task_bwa.wdl" as bwa_task
+import "../../tasks/assembly/task_ivar_consensus.wdl" as ivar_consensus_task
+import "../../tasks/gene_typing/variant_detection/task_ivar_variant_call.wdl" as variant_call_task
+import "../../tasks/quality_control/basic_statistics/task_assembly_metrics.wdl" as assembly_metrics_task
+import "../../tasks/quality_control/basic_statistics/task_consensus_qc.wdl" as consensus_qc_task
+import "../../tasks/task_versioning.wdl" as versioning_task
+import "../../workflows/utilities/wf_morgana_magic.wdl" as morgana_magic_wf
+
+workflow theiaviral_illumina_pe {
+  meta {
+    description: "De novo assembly, dynamic reference selection, and reference-based consensus calling for viral metagenomic/genomic data generated on Illumina paired-end NGS platforms."
+  }
+  input {
+    File read1
+    File read2
+    String taxon # taxon id OR organism name
+    String read_extraction_rank = "family"
+    String samplename
+    String? host # host to dehost reads, if provided
+    File kraken_db = "gs://theiagen-public-resources-rp/reference_data/databases/kraken2/kraken2_humanGRCh38_viralRefSeq_20240828.tar.gz"
+    Boolean skip_screen = false # if false, run clean read screening
+    Boolean call_metaviralspades = true # if false, move to megahit immediately
+    Boolean skip_rasusa = true 
+    File? reference_fasta # optional, if provided, will be used instead of dynamic reference selection
+    Boolean extract_unclassified = false # if true, unclassified reads will be extracted from kraken2 output
+    Int min_depth = 10
+    Int min_map_quality = 20
+    Float min_allele_freq = 0.6
+    # rasusa downsampling inputs
+    Int? genome_length
+  }
+  # get the PHB version
+  call versioning_task.version_capture {
+    input:
+  }
+  # get the taxon id
+  call identify_taxon_id_task.identify_taxon_id as ncbi_identify {
```

Comment:
can you combine ncbi_identify with ncbi_datasets_genome_summary? You can pass in genome_length so that the summary section only runs if genome_length is not provided. This will save in runtime and cost since only 1 VM will be spun up when both tasks are using the same docker image.